### PR TITLE
Remove authoritative option

### LINKED ISSUE #3640: The '-A' switch of 'complete' command is not documented
The `-A` switch of the `complete` command

is used
```console
$ grep -r complete share/completions/ | grep " \-A " | wc -l
35
```
but not documented
```
$ grep " \-A " doc_src/complete.txt | wc -l
0
```

### PR DESCRIPTION
**Pull request by radomirbosak**
## Remove --authoritative leftovers from complete

The complete builtin had once `-A` / `--authoritative` and `-u` /
`--unauthoritative` switches which indicated whether all possibilities for
completion are specified and would cause an error if the completion was
authoritative and an unknown option was encountered.

This feature was functionally removed during one of the past parser
rewritings, but `-A` and `-u` still remained in parts of the code and
command completions, although having no effect.

This pull request removes the leftovers and prints an warning whenever user
tries to run the complete command with `-A` / `-u` / `--authoritative` /
`--unauthoritative` switches. Moreover, all mentions of these switches are removed from fish completions.

Fixes #3640.

### CODE REVIEW

File: src/builtin_complete.cpp [hunk 1]
```diff
@@ -193,11 +184,11 @@ int builtin_complete(parser_t &parser, io_streams_t &streams, wchar_t **argv) {
                 break;
             }
             case 'u': {
-                authoritative = 0;
+                streams.err.append_format(L"%ls: -u / --unauthoritative flags have been removed\n", cmd);
```
Comment (krader1961) [inline src/builtin_complete.cpp]: Messages are normally wrapped by the localization function `_()`.

Comment (krader1961) [inline src/builtin_complete.cpp]: Ahh, I see you were just following the pattern of the surrounding code. I'll fix that when I merge your change.

**Comment by krader1961** [comment]
Merged as 720c0ddb3, 2900352e0, and dc96c01c9. Thanks.

**Comment by zanchey** [comment]
Yow; this is a pretty obnoxious change (I get hundreds of lines of "complete: -A / --authoritative flags have been removed" when I go to tab-complete). As this is is effectively a no-op change, can I suggest silently ignoring the flag (perhaps with a note in the man page to note they are deprecated and ignored for compatibility), or at least silence the errors if my completions are being autoloaded?

**Comment by floam** [comment]
I think we should have waited a little while before merging this. It's really noisy - the PR should have at least cleaned that up.

And are we so sure we don't like the idea of an -A option? It used to be cool that fish would mark a line as an error if the arguments were not legal.

**Comment by krader1961** [comment]
I still think a warning is appropriate to alert people to the fact they may be relying on functionality that is no longer present. However, we should limit it to exactly one warning per session.

**Comment by ridiculousfish** [comment]
It sounds like we should back this out. All `authoritative` ever did was affect syntax highlighting, so it's hard to see how anyone could be relying on it This is a breaking change, and the warning isn't really helping anyone avoid any problems.

**Comment by krader1961** [comment]
We definitely should not revert the entire change. Removing use of the obsolete options from our completions is a good thing. Whether we should issue a warning when the obsolete option is seen needs more debate. Especially in light of commit 7a3e0924 I made to limit fish to printing exactly one such warning per session.

If we do print a warning with this release it draws a line in the sand. When at least two years have gone by we can reuse those flags. If we don't print a warning we'll never be able to reuse those flags. So the question is whether we ever want to be able to reuse those flags and whether that desire outweighs the minor annoyance of a single warning due to third-party completions that  haven't been updated.

**Comment by zanchey** [comment]
It's a warning that users may have no control over, and it may not be actionable. I don't think "drawing a line in the sand" is that helpful; the entire completion description system may well change in the future (e.g. docopt), in which case the `complete` builtin is going to become much less important. I'd like the warning dropped.

**Comment by krader1961** [comment]
Okay, I've pushed commit f365f72 which drops the warning and updates the man page to list them as being no-ops.

**Pull request closed without merging**